### PR TITLE
Add Spark 3.4 support

### PR DESCRIPTION
**Pull request by jhchee**
*Issue #, if available:*
https://github.com/awslabs/deequ/issues/481 

*Description of changes:*
- Add Spark 3.4 support 

`mvn clean package` run successfully on my local machine. Test result is as followed:
```
Run completed in 4 minutes, 56 seconds.
Total number of tests run: 552
Suites: completed 64, aborted 0
Tests: succeeded 552, failed 0, canceled 0, ignored 0, pending 0
All tests passed.
```
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -18,7 +18,7 @@
         <artifact.scala.version>${scala.major.version}</artifact.scala.version>
         <scala-maven-plugin.version>4.8.1</scala-maven-plugin.version>
 
-        <spark.version>3.3.0</spark.version>
+        <spark.version>3.4.1</spark.version>
```
Comment (jhchee) [inline pom.xml:21]: I have added 3.4.1 as this is the latest version in the release line.

File: pom.xml [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
     <groupId>com.amazon.deequ</groupId>
     <artifactId>deequ</artifactId>
-    <version>2.0.3-spark-3.3</version>
+    <version>2.0.4-spark-3.4</version>
```
Comment (jhchee) [inline pom.xml:9]: I have incremented the build version. Let me know if this should be separated into different PR.

**Comment by jhchee** [comment]
@eycho-am Do you mind to have a look? Spark 3.4 has been released for quite for time already and Spark 3.5 is just around the corner.

**Review by rdsharma26 — APPROVED** [review]
